projects/go-lib/src/lib/components/go-timepicker/go-timepicker.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { generateId } from '../../utilities/form.utils';
+import { GoTime } from './go-time';
+@Component({
+  selector: 'go-timepicker',
+  styleUrls: ['./go-timepicker.component.scss'],
+  templateUrl: './go-timepicker.component.html',
+})
+export class GoTimepickerComponent implements OnInit {
+  displayAbove: boolean = false;
+  displayFromRight: boolean = true;
+  id: string;
+
+  @Input() key: string;
+  @Input() label: string;
+  @Input() control: FormControl;
+  @Input() hints: string[];
+  @Input() placeholder: string = '';
+  @Input() theme: string = 'light';
+  @Input() appendToContent: boolean = false;
+
+  goTime: GoTime;
+
+  selectedTime: string;
+
+  @ViewChild('datepickerInput', { static: true }) datepickerInput: ElementRef;
+  constructor() {
+    this.goTime = new GoTime();
+  }
+  ngOnInit(): void {
+    this.id = this.key || generateId(this.label, 'timepicker');
+    this.selectedTime = this.changeTimeFormat(this.control.value);
+  }
+  public changeTimeFormat(timeString) {
+    if (!timeString) {
+      return;
+    }
+    const H = +timeString.substr(0, 2);
+    const h = H % 12 || 12;
+    const hour = h > 9 ? h : '0' + h;
+    const ampm = H < 12 || H === 24 ? 'AM' : 'PM';
+    timeString = hour + timeString.substr(2, 3) + ' ' + ampm;
+    return timeString;
+  }
+
+  public validateTime(): void {}
```

Comment:
Are we using this method?

Reply:
No, I had a plan to use it. But now it's not required. I have removed it.